aten/src/THCUNN/SpatialUpSamplingNearest.cu

```diff
@@ -2,85 +2,101 @@
 #include "common.h"
 #include "THCTensor.hpp"
 
-#include <thrust/transform.h>
-#include <thrust/reduce.h>
-#include <thrust/transform_reduce.h>
-#include <thrust/functional.h>
+#include "linear_upsampling.h"
+#include "THCDeviceTensor.cuh"
+#include "THCDeviceTensorUtils.cuh"
+#include "THCDeviceUtils.cuh"
 
 #include "THCHalf.h"
 #include "THCHalfAutoNumerics.cuh"
+#include "THCAtomics.cuh"
 
-/*
- * Description:
- */
+template<typename Dtype, typename Acctype>
```

Comment:
maybe add a comment saying that this kernel is based on the TF implementation, with the link to the implementation?

Reply:
Our implementation is not exactly the same as TF as they also have align_corners in the implementation, which doesn't make sense. Not sure if we still want to put the link?